akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/DynamoDBProjectionImpl.scala

```diff
@@ -318,30 +404,31 @@ private[projection] object DynamoDBProjectionImpl {
       override def process(envelopes: Seq[Envelope]): Future[Done] = {
         import DynamoDBOffsetStore.Validation._
         offsetStore.validateAll(envelopes).flatMap { isAcceptedEnvelopes =>
+          // FIXME improvement would be to only replay the last for each pid
           val replayDone =
-            Future.sequence(isAcceptedEnvelopes.map {
-              case (env, RejectedSeqNr) =>
-                triggerReplayIfPossible(sourceProvider, offsetStore, env).map(_ => Done)(ExecutionContext.parasitic)
-              case (env, RejectedBacktrackingSeqNr) =>
-                triggerReplayIfPossible(sourceProvider, offsetStore, env).map {
-                  case true  => Done
-                  case false => throwRejectedEnvelope(sourceProvider, env)
+            isAcceptedEnvelopes.foldLeft(FutureDone) {
+              case (previous, (env, RejectedSeqNr)) =>
+                previous.flatMap { _ =>
+                  replayIfPossible(env).map(_ => Done)(ExecutionContext.parasitic)
```

Comment:
It might feel strange at first to replay the rejected first, and then process the accepted, but as far as I can see that should be fine. The order per pid should still be fine, and it will replay from previously stored seqNr so that could cover some of the accepted as well for that pid. I added a second validation step before processing the accepted to remove such duplicates.

Reply:
Yeah, saw that was already there and seemed to make sense. And then second validation.